packages/react-focus/src/components/FocusZone/FocusZone.tsx

```diff
@@ -469,6 +463,12 @@ export class FocusZone extends React.Component<IFocusZoneProps> implements IFocu
     if (stopFocusPropagation || doNotAllowFocusEventToPropagate) {
       ev.stopPropagation();
     }
+
+    if (onFocus) {
+      onFocus(ev);
```

Comment:
My thought would be that you do the user call first. 

Then, if they have called ev.preventDefault(), don't do anything additional.

```jsx
onFocus?.(ev);

if (!ev.defaultPrevented) {
  ... do stuff
}
```